kolibri/plugins/coach/assets/src/views/common/QuizStatus.vue

```diff
@@ -0,0 +1,248 @@
+        if (this.exam.date_activated) {
+          return new Date(this.exam.date_activated);
+        } else {
+          return null;
+        }
+      },
+    },
+    methods: {
+      handleOpenQuiz() {
+        let promise = ExamResource.saveModel({
+          id: this.$route.params.quizId,
+          data: {
+            active: true,
+            date_activated: new Date(),
+          },
+          exists: true,
+        });
+
+        return promise
+          .then(() => {
+            this.$store.dispatch('classSummary/refreshClassSummary');
+            this.showConfirmationModal = false;
+            this.$store.dispatch('createSnackbar', this.coachString('quizOpenedMessage'));
+          })
+          .catch(() => {
+            this.$store.dispatch('createSnackbar', this.coachString('quizFailedToOpenMessage'));
+          });
+      },
+      handleCloseQuiz() {
+        let promise = ExamResource.saveModel({
+          id: this.$route.params.quizId,
+          data: {
+            archive: true,
+            date_archived: new Date(),
+          },
+          exists: true,
+        });
+
+        return promise
+          .then(() => {
+            this.$store.dispatch('classSummary/refreshClassSummary');
+            this.showCancellationModal = false;
+            this.$store.dispatch('createSnackbar', this.coachString('quizClosedMessage'));
+          })
+          .catch(() => {
+            this.$store.dispatch('createSnackbar', this.coachString('quizFailedToCloseMessage'));
+          });
+      },
+      handleToggleVisibility() {
+        const newActiveState = !this.exam.active;
+        const snackbarMessage = newActiveState
+          ? this.coachString('quizVisibleToLearners')
+          : this.coachString('quizNotVisibleToLearners');
+
+        let promise = ExamResource.saveModel({
+          id: this.$route.params.quizId,
+          data: {
+            active: newActiveState,
+          },
+          exists: true,
+        });
+
+        return promise.then(() => {
+          this.$store.dispatch('classSummary/refreshClassSummary');
+          this.showConfirmationModal = false;
+          this.$store.dispatch('createSnackbar', snackbarMessage);
+        });
+      },
+    },
+    $trs: {
+      quizClosedLabel: {
+        message: 'Quiz closed',
+        context:
+          'A label indicating that the currently viewed quiz is closed - meaning that learners may no longer give answers to the quiz.',
+      },
+      reportVisibleToLearnersLabel: {
+        message: 'Report visible to learners',
+        context:
+          'The label for a switch that will toggle whether or not learners can view their quiz report.',
+      },
+      questionOrderLabel: {
+        message: 'Question order',
+        context: 'A label for the place where the question order is shown.',
+      },
+      yesLabel: {
+        message: 'Yes',
+        context: "Used to indicate to the user when a switch element has a true or 'on' value.",
```

Comment:
Indicates the value of the switch element turned 'on', and the quiz report is visible for the learners.

Reply:
Shouldn't there be a `No` value/message here, instead of `Yes`? Toggle is turned off... :confused: 

cc @jtamiace 

![LEDev1911n  Running  - Oracle VM VirtualBox_023](https://user-images.githubusercontent.com/1457929/67449510-b2271880-f61a-11e9-81fb-f0c8ba3b098a.png)

`<rant-on>`
_I really, **really**, dislike toggles... Nothing but confusion_ :unamused: 
`<rant-off>`

Reply:
@radinamatic I asked @jtamiace and @khangmach about this while working on it and the standard seems to be that the visual color change largely plays a role in indicating status.

The "Yes" is there to indicate which side of the switch is the "Yes" position - I presume to cover cases of color-blindness. I initially thought that it should be a visual representation of the current value - but that's not the typical use of the switch.

Reply:
>  I initially thought that it should be a visual representation of the current value

That is correct. From the design:

![image](https://user-images.githubusercontent.com/2367265/67501349-fb866080-f638-11e9-869a-c0ac08a7c05d.png)

Reply:
Ok, so there must be a `No` value/message visible to learners when the toggle is OFF, as in the image above, that's a relief :relieved:

Reply:
@indirectlylit @radinamatic 

I mentioned this to @jtamiace in a Slack DM but should have probably opened it in #design - anyway, here is her message on the topic:

> think we came to the conclusion that the label should always say “yes,” and leave the toggle as the thing that communicates current system status to the user
there are issues with the label being dynamic that outlined here https://www.nngroup.com/articles/toggle-switch-guidelines/
this descriptor is ambiguous: because the word Off appears on the right of the switch and the toggle dot is on the left, users can also interpret the state of the toggle as on (that is, away from the label Off).

Reply:
To me this is still equally and utterly confusing: two different visually perceivable UI element states display the same "label":..?

![confusing](https://user-images.githubusercontent.com/1457929/67521509-50fb5580-f6ab-11e9-8e83-f966ce718da7.png)

Compounding issue is that fusing the two possible actions (open/close quiz: toggle report visibility) in the same column, and eliminating the value of the column header, makes it inaccessible.

Reply:
resolved in https://www.notion.so/learningequality/Final-behaviors-of-switch-toggles-in-0-13-lesson-quiz-updates-dec8f9feeaa54e6cb5122605b0a9b32f

- Switches no longer have labels in either lessons or quizzes, resulting in a slight change in layout
- Removed the switch toggle for report visibility from the quiz report table, so there is only a "Status" column that show actions and status: "Open quiz," "Close quiz," "Quiz closed"
    - Reason: confusing to have "open/close quiz" & "show/hide report" as they have overlaps in status visibility. There was also no clear user need to display the report switch in the table so we decided to return to that issue later if there is feedback on it.